consensus/misc/eip1559.go

```diff
@@ -33,11 +35,35 @@ const MaximumL2BaseFee = 10000000000
 // `scalar` accounts for finalization costs. `overhead` accounts for sequencing and proving costs.
 // we use 1e18 for precision to match the contract implementation.
 var (
-	BaseFeePrecision       = new(big.Int).SetUint64(1e18)
-	DefaultBaseFeeScalar   = new(big.Int).SetUint64(34000000000000)
-	DefaultBaseFeeOverhead = new(big.Int).SetUint64(15680000)
+	BaseFeePrecision = new(big.Int).SetUint64(1e18)
+
+	// scalar and overhead are updated automatically in `Blockchain.writeBlockWithState`.
+	baseFeeScalar   = big.NewInt(0)
+	baseFeeOverhead = big.NewInt(0)
+
+	lock sync.RWMutex
 )
 
+func UpdateL2BaseFeeOverhead(newOverhead *big.Int) {
+	if newOverhead == nil {
+		log.Error("Failed to set L2 base fee overhead, new value is <nil>")
+		return
+	}
+	lock.Lock()
+	defer lock.Unlock()
+	baseFeeOverhead.Set(newOverhead)
+}
+
+func UpdateL2BaseFeeScalar(newScalar *big.Int) {
+	if newScalar == nil {
```

Comment:
So does this (normally won't happen unless some unexpected bug). Maybe a practical solution would be to check error logs during updating params?